src/passes/OptimizeInstructions.cpp

```diff
@@ -1005,6 +1006,98 @@ struct OptimizeInstructions
     }
   }
 
+  void visitRefIs(RefIs* curr) {
+    if (curr->type == Type::unreachable) {
+      return;
+    }
+
+    // Optimizating RefIs is not that obvious, since even if we know the result
+    // evaluates to 0 or 1 then the replacement may not actually save code size,
+    // since RefIsNull is a single byte (the others are 2), while adding a Const
+    // of 0 would be two bytes. Other factors are that we can remove the input
```

Comment:
Does this include `drop` in the calculation too? For example, in `RefIsNull` case, we save one byte by removing `ref.is_null`, add one byte by `drop`, and add two more bytes for `i32.const x`, so isn't that net +2 bytes gain? (+1 byte gain for other `ref.is_***` instructions)

Reply:
I should have mentioned the drop too, yes. In most cases the input value has no side effects so both it and the drop for it can be removed.